### PR TITLE
ResDiff integration: Add dependencies

### PR DESCRIPTION
**Pull request by mnabian**
# Modulus Pull Request

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

**Review by MortezaMardani — APPROVED** [review]

**Review by ktangsali — APPROVED** [review]

File: pyproject.toml
Comment (ktangsali) [inline pyproject.toml:1]: Let's add a comment to CHANGELOG since this is changing dependencies. Thanks!

Comment (mnabian) [reply pyproject.toml:1]: Sure, done

**Pull request merged**